packages/teams-js/src/public/appNotification.ts

```diff
@@ -0,0 +1,182 @@
+export namespace appNotification {
+  /**
+   * Data structure to represent appNotification information
+   */
+  export interface NotificationDisplayParam {
+    /**
+     * Notification title(maximum length: 20 characters)
+     */
+    title: string;
+    /**
+     * Notification content (maximum length: 40 characters)
+     */
+    content: string;
+    /**
+     * This would represent an optional icon that can be displayed on the notification. It should have a max size of 49 pixels by 49 pixels
+     * If no icon is provided, the notification card would be displayed without an icon
+     * The url link to where the icon is stored should be provided as the input string
+     */
+    icon?: string;
+    /**
+     * This would specify how long a notification would be displayed on the screen for (unit: seconds)
+     *
+     */
+    displayDurationInSeconds: number;
+    /**
+     * A url link to the page in which the notification would direct the user to.
+     */
+    notificationActionUrl: URL;
+  }
+
+  /**
+   * Data structure with simplified parameters that would be passed across the iframe
+   */
+  export interface NotificationDisplayParamForAppHost {
+    /**
+     * Notification title(maximum length: 20 characters)
+     */
+    title: string;
+    /**
+     * Notification content (maximum length: 40 characters)
+     */
+    content: string;
+    /**
+     * This would represent an optional icon that can be displayed on the notification. It should have a max size of 49 pixels by 49 pixels
+     * If no icon is provided, the notification card would be displayed without an icon
+     * The url link to where the icon is stored should be provided as the input string
+     */
+    icon?: string;
+    /**
+     * This would specify how long a notification would be displayed on the screen for (unit: seconds)
+     *
+     */
+    displayDurationInSeconds: number;
+    /**
+     * A url string type to the page in which the notification would direct the user to.
+     */
+    notificationActionUrlAsString: string;
+  }
+
+  /**
+   * This converts the notifcationActionUrl from a URL type to a string type for proper flow across the iframe
+   * @param ndParam
+   * @returns a NotificationDisplay object with a string type parameter for the notificationAtionUrl
+   */
+  function serializeParam(notificationDisplayParam: NotificationDisplayParam): NotificationDisplayParamForAppHost {
+    return {
+      title: notificationDisplayParam.title,
+      content: notificationDisplayParam.content,
+      icon: notificationDisplayParam.icon,
+      displayDurationInSeconds: notificationDisplayParam.displayDurationInSeconds,
+      notificationActionUrlAsString: notificationDisplayParam.notificationActionUrl.href,
+    };
+  }
+
+  /**
+   * Checks the valididty of the URL passed
+   *
+   * @param url
+   * @returns True if a valid url was passed
+   */
+  function isValidUrl(url: URL): boolean {
+    const validProtocols = ['http:', 'https:', 'ftp:'];
```

Comment:
>'http:', 'https:', 'ftp:' [](http://example.com/codeflow?start=28&length=25)

Are ftp sites really valid for this? Should we restrict to https sites for privacy/security reasons?

Reply:
Okay, Are you also suggesting taking off the http as well?